### ISSUE TITLE
[FEATURE] Add a proxmox_virtual_environment_vm datasource

### ISSUE THREAD

**Issue by Zempashi** [danitso/terraform-provider-proxmox#106]
**Is your feature request related to a problem? Please describe.**
Provider return ```Error: Received an HTTP 500 response - Reason: unable to find configuration file for VM 9000 on node 'pve-03', because I omitted the `node_name` during clone (I provided just `vm_id`: 9000 not located on the node I want to put my VM)                                                                                         

**Describe the solution you'd like**
A datasource `proxmox_virtual_environment_vm` would be nice to find a VM by it's id or by it's name, so we can found node_name and vm_id we can pass later to clone subresource.

**Describe alternatives you've considered**
Other possibility like the telmate/terraform-provider-proxmox, would be to support clone by it's name directly, but I like the philosophy of this provider where resource are clear and no magic happens, so I found datasource would offer more use cases...

**Comment by Anguianolabs** [comment]
This project has died.

**Comment by dmeulen** [comment]
Yup, such a pity IMHO this provider is much better than the telemate provider. Don't know what happened here, all danitso's projects are suddenly flatlined ( hope the guy is still alive and kicking ). But the `bpg` fork seems to implemt some fixes...

**Comment by danitso-dp** [comment]
The project is on hiatus ;-)

No. Jokes aside, I've been busy with life and my full time job. I simply haven't had time to continue my work on this project. I haven't abandoned it but it's hard to find the time for it right now. Hopefully, that'll change.
